src/authInfo.ts

```diff
@@ -500,6 +501,79 @@ export class AuthInfo extends AsyncCreatable<AuthInfo.Options> {
     };
   }
 
+  /**
+   * Given a set of decrypted fields and an authInfo, determine if the org belongs to an available
+   * dev hub.
+   *
+   * @param fields
+   * @param orgAuthInfo
+   */
+  public static async identifyPossibleScratchOrgs(fields: AuthFields, orgAuthInfo: AuthInfo): Promise<void> {
+    // fields property is passed in because the consumers of this method have performed the decrypt.
+    // This is so we don't have to call authInfo.getFields(true) and decrypt again OR accidentally save an
+    // authInfo before it is necessary.
+    const logger = await Logger.child('Common', { tag: 'identifyPossibleScratchOrgs' });
+
+    // return if we already know the hub org we know it is a devhub or prod-like or no orgId present
+    if (fields.isDevHub || fields.devHubUsername || !fields.orgId) return;
+    // there are no hubs to ask, so quit early
+    if (!(await AuthInfo.hasAuthentications())) return;
+    logger.debug('getting devHubs from authfiles');
+
+    // TODO: return if url is not sandbox-like to avoid constantly asking about production orgs
+    // TODO: someday we make this easier by asking the org if it is a scratch org
+
+    const hubAuthInfos = await this.getDevHubAuthInfos();
+    logger.debug(`found ${hubAuthInfos.length} DevHubs`);
+    if (hubAuthInfos.length === 0) return;
+
+    // ask all those orgs if they know this orgId
+    await Promise.all(
+      hubAuthInfos.map(async (hubAuthInfo) => {
+        try {
+          const data = await AuthInfo.queryScratchOrg(hubAuthInfo.getUsername(), fields.orgId as string);
```

Comment:
could save _some_ iterations by setting `const hubAuthUsername = hubAuthInfo.getUsername()` instead of calling it 3+ times in a row. Especially since this is in a `Promise.all`

Reply:
@WillieRuemmele `hubAuthInfo.getUsername()` is a simple getter. I could see running it once if it were async or or function was very complex.